src/realm/impl/array_writer.hpp

```diff
@@ -30,6 +31,7 @@ class ArrayWriterBase {
     bool only_modified = true;
     bool compress = true;
     const Table* table;
+    ColKey col_key;
```

Comment:
Why do you need the col_key here? Doesn't the table contain all the information you need (1 string interner per col key).

Reply:
In ArrayString we need the ColKey in order to get the interner.